test.py

```diff
@@ -343,7 +343,11 @@ def __init__(self, path, cfg: dict, options: argparse.Namespace, mode: str) -> N
         pool_size = cfg.get("pool_size", 2)
 
         self.create_cluster = self.get_cluster_factory(cluster_size)
-        self.clusters = Pool(pool_size, self.create_cluster)
+        async def destroy_cluster(cluster: ScyllaCluster) -> None:
```

Comment:
Please provide a docstring. I think it's not a complete destruction, since it leaves the data dir around. Perhaps some other name, like recycle()?